README.md

```diff
@@ -1,34 +1,25 @@
-# node-controller-manager
+# Node-Controller-Manager
 
-The Node Controller Manager bundles multiple Kubernetes controllers, that help with the creation, maintance and operation of different instance (node) Kubernetes resources. 
+Node Controller Manager (NCM) manages VMs as another kubernetes custom resource. It provides a declarative way to manage VMs. The current implementation only supports AWS but can be (and will be) easily extended to support other cloud providers.
 
-It's designed to run in master plane of a kubernetes cluster and follows the best principles and practices of writing controllers, including, but not limited to:
-
-- leader election to allow HA deployments of the controller
-- `workqueues` and multiple thread-workers
-- `SharedInformers` that limit to minimum network calls, deserialization and provide helpful create/update/delete events for resources
-- rate-limiting to allow back-off in case of network outages and general instability of other cluster components
-- sending events to respected resources for easy debugging and overview
-- Prometheus metrics, health and (optional) profiling endpoints
-
-
-
-## Build
-
-```bash 
-go build -i cmd/node-controller-manager/controller_manager.go
+Example of managing VM:
 ```
+kubectl create/get/delete instance vm1
+``` 
 
-## Running
+### Key terminologies
 
-For testing/development purposes, you can create your own kubernetes cluster and download its kubeconfig file. Use this kubeconfig file to be passed as an argument to your node-controller-manager. You can run the node-controller-manager on your local machine as mentioned below
+Nodes/Instances/Machines/VMs are different terminologies used to represent the similar things in theory. However, for most of our discussion the following are our interpretations.
 
-```bash
-go run cmd/node-controller-manager/controller_manager.go --kubeconfig=$KUBECONFIG --v=2
-```
+1. VM: A virtual machine running on any cloud provider.
+1. Node: Native kubernetes node objects. The objects you get to see when you do a *"kubectl get nodes"*. Although nodes can be either physical/virtual machines, for the purposes of our discussions it refers to a VM. 
+1. Instance: A VM that is provisioned/managed by the Node Controller Manager.
```

Comment:
why don't you adopt the term machine used by the community right away ?

Reply:
Yes, we are doing it soon in the next PR.